modules/adfs/lib/IdP/ADFS.php

```diff
@@ -157,7 +164,7 @@ public static function sendResponse(array $state) {
 		$wresult = sspmod_adfs_IdP_ADFS::ADFS_SignResponse($response, $privateKeyFile, $certificateFile);
 
 		$wctx = $state['adfs:wctx'];
-		sspmod_adfs_IdP_ADFS::ADFS_PostResponse($spMetadata->getValue('prp'), $wresult, $wctx);
+		sspmod_adfs_IdP_ADFS::ADFS_PostResponse(($state['adfs:wreply']) ? $state['adfs:wreply'] : $spMetadata->getValue('prp'), $wresult, $wctx);
```

Comment:
Here we should be explicit, I think. Either by calling `isset($state['adfs:wreply'])` if that key is only set conditionally, or by setting it to false if no `wreply` parameter is present.